src/runtime_src/core/pcie/driver/linux/xocl/subdev/m2m.c

```diff
@@ -0,0 +1,223 @@
+// SPDX-License-Identifier: GPL-2.0
+/*
+ * Xilinx CU driver for memory to memory BO copy
+ *
+ * Copyright (C) 2020 Xilinx, Inc.
+ *
+ * Authors: David Zhang <davidzha@xilinx.com>
+ */
+
+#include <linux/types.h>
+#include <linux/uaccess.h>
+#include <linux/vmalloc.h>
+#include <linux/slab.h>
+#include <linux/cdev.h>
+#include <linux/fs.h>
+#include <linux/device.h>
+#include <linux/io.h>
+#include <linux/ioctl.h>
+
+#include "../xocl_drv.h"
+#include "../userpf/common.h"
+#include "xrt_cu.h"
+
+/* IOCTL interfaces */
+#define XOCL_M2M_MAGIC 0x4d324d // "M2M"
+
+#define	M2M_ERR(m2m, fmt, arg...)	\
+	xocl_err(&(m2m)->m2m_pdev->dev, fmt "\n", ##arg)
+#define	M2M_WARN(m2m, fmt, arg...)	\
+	xocl_warn(&(m2m)->m2m_pdev->dev, fmt "\n", ##arg)
+#define	M2M_INFO(m2m, fmt, arg...)	\
+	xocl_info(&(m2m)->m2m_pdev->dev, fmt "\n", ##arg)
+#define	M2M_DBG(m2m, fmt, arg...)	\
+	xocl_dbg(&(m2m)->m2m_pdev->dev, fmt "\n", ##arg)
+
+/* This is the real register map for the copy bo CU */
+struct start_copybo_cu_cmd {
+  uint32_t src_addr_lo;      /* low 32 bit of src addr */
+  uint32_t src_addr_hi;      /* high 32 bit of src addr */
+  uint32_t src_bo_hdl;       /* src bo handle, cleared by driver */
+  uint32_t dst_addr_lo;      /* low 32 bit of dst addr */
+  uint32_t dst_addr_hi;      /* high 32 bit of dst addr */
+  uint32_t dst_bo_hdl;       /* dst bo handle, cleared by driver */
+  uint32_t size;             /* size in bytes */
```

Comment:
This register should be size in bus width. In current design, the bus width is 512 bits (64 bytes). Please define a macro "#define KDMA_DATA_WIDTH 64".

Reply:
Already have "KDMA_BLOCK_SIZE" in ert.h, no need to define a new macro.
Please only correct this comment.

Reply:
adjusted the comments